velPaint

```diff
@@ -85,20 +89,17 @@ while (my $line = <>) {
 	chomp $line;
 	$line =~ s/\r//g;
 
-	($x, $y, $z, $e, $f) =
-	 $line =~ /G1 X([^\s]+) Y([^\s]+) Z([^\s]+) E([^\s]+) F($targetSpeed)$/;
-	($x, $y, $e, $f) =
-	 $line =~ /G1 X([^\s]+) Y([^\s]+) E([^\s]+) F($targetSpeed)$/ unless (defined $x);
-	($x, $y, $z, $e) =
-	 $line =~ /G1 X([^\s]+) Y([^\s]+) Z([^\s]+) E([^\s]+)$/ unless (defined $x);
-	($x, $y, $e) =
-	 $line =~ /G1 X([^\s]+) Y([^\s]+) E([^\s]+)$/ unless (defined $x);
+	if($line =~ /G1 X(?<x>\S+) Y(?<y>\S+)(?: Z(?<z>\S+))? E(?<e>\S+)(?: F(?<f>$targetSpeed))?$/ ||
+	  $line =~ /G1 Z(?<z>\S+) X(?<x>\S+) Y(?<y>\S+) E(?<e>\S+)(?: F(?<f>$targetSpeed))?$/)
+	{
+		($x, $y, $z, $e, $f) = @+{qw/x y z e f/};
+	}
 
 	if (defined $z) { $currentZ = $z } else { $z = $currentZ; }
 
 	if (defined $x) {
 		if (!defined $oldZ) {
-			&outMove($x, $y, $z, $e, 0);
+			outMove($x, $y, $z, $e, 0);
```

Comment:
Calling subs using the `&` sigil is not recommended. It's a practice from perl4 and it can have interesting side effects in perl5 under some circumstances. (None of which apply in this script.)